dwa_local_planner/src/dwa_planner.cpp

```diff
@@ -375,14 +379,17 @@ namespace dwa_local_planner {
     // costs for robot being aligned with path (nose on path)
     alignment_costs_.setTargetPoses(global_plan_);
 
+    // costs for coming towards goal at different heading
+    goal_heading_costs_.setGoalPose(global_plan_.back());
 
     // keeping the nose on the path
     if (sq_dist > close_to_goal_range_ * close_to_goal_range_) {
       double resolution = planner_util_->getCostmap()->getResolution();
       alignment_costs_.setScale(resolution * pdist_scale_ * 0.5);
 
       goal_front_costs_.setScale(resolution * gdist_scale_ * 0.5);
-      goal_costs_.setScale(resolution * gdist_scale_ * 0.5);
+      goal_costs_.setScale(resolution * gdist_scale_ * 0.5); // what are these scales
```

Comment:
Scales are the relative weights.

Reply:
what's `resolution` and why multiply by 0.5?